src/View/CastingService.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace SilverStripe\View;
+
+use LogicException;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Model\ArrayData;
+use SilverStripe\Model\List\ArrayList;
+use SilverStripe\Model\ModelData;
+use SilverStripe\ORM\FieldType\DBBoolean;
+use SilverStripe\ORM\FieldType\DBFloat;
+use SilverStripe\ORM\FieldType\DBInt;
+use SilverStripe\ORM\FieldType\DBText;
+
+class CastingService
+{
+    use Injectable;
+
+    /**
+     * Cast a value to the relevant object (usually a DBField instance) for use in the view layer.
+     *
+     * @param null|array|ModelData $source Where the data originates from. This is used both to check for casting helpers
+     * and to help set the value in cast DBField instances.
+     * @param bool $strict If true, an object will be returned even if $data is null.
+     */
+    public function cast(mixed $data, null|array|ModelData $source = null, string $fieldName = '', bool $strict = false): ?object
```

Comment:
Change `null|array|ModelData` to `ModelData|array|null` - looks weird having null first
Change `$strict` to `$force`

Reply:
That's..... just super subjective. I'll change it to avoid ping pong but on a large PR like this having subjective stylistic changes extends the review cycle without any objective improvement.

I'd recommend avoiding subjective stylistic suggestions like this or else raising a card to introduce a standard for this that we can objectively follow.

Reply:
OK leave as is